src/Common/ThreadStatus.cpp

```diff
@@ -237,6 +237,7 @@ void ThreadStatus::setFatalErrorCallback(std::function<void()> callback)
 
 void ThreadStatus::onFatalError()
 {
+    std::lock_guard lock(thread_group->mutex);
```

Comment:
@alexey-milovidov, `thread_group` may not be set: https://pastila.nl/?02a69da9/edd0fb1660431e9ce76569b799c52032